packages/core/test/unit/stryker.spec.ts

```diff
@@ -197,4 +201,50 @@ describe(Stryker.name, () => {
       expect(shutdownLoggingStub).called;
     });
   });
+
+  describe('runMutationTest()', () => {
+    beforeEach(() => {
+      injectorMock = factory.injector();
+      loggerMock = factory.logger();
+      strOptions = factory.strykerOptions({ cleanTempDir: 'always' });
+      getLoggerStub = sinon.stub();
+      mutantResults = [];
+      temporaryDirectoryMock = sinon.createStubInstance(TemporaryDirectory);
+      prepareExecutorMock = sinon.createStubInstance(PrepareExecutor);
+      mutantInstrumenterExecutorMock = sinon.createStubInstance(MutantInstrumenterExecutor);
+      dryRunExecutorMock = sinon.createStubInstance(DryRunExecutor);
+      mutationTestExecutorMock = sinon.createStubInstance(MutationTestExecutor);
+      injectorMock.injectClass
+        .withArgs(PrepareExecutor)
+        .returns(prepareExecutorMock)
+        .withArgs(MutantInstrumenterExecutor)
+        .returns(mutantInstrumenterExecutorMock)
+        .withArgs(DryRunExecutor)
+        .returns(dryRunExecutorMock)
+        .withArgs(MutationTestExecutor)
+        .returns(mutationTestExecutorMock);
+      injectorMock.resolve
+        .withArgs(commonTokens.getLogger)
+        .returns(getLoggerStub)
+        .withArgs(coreTokens.temporaryDirectory)
+        .returns(temporaryDirectoryMock)
+        .withArgs(commonTokens.options)
+        .returns(strOptions);
+      getLoggerStub.returns(loggerMock);
+
+      prepareExecutorMock.execute.resolves(injectorMock as typedInject.Injector<MutationTestContext>);
+      mutantInstrumenterExecutorMock.execute.resolves(injectorMock as typedInject.Injector<MutationTestContext>);
+      dryRunExecutorMock.execute.resolves(injectorMock as typedInject.Injector<MutationTestContext>);
+      mutationTestExecutorMock.execute.resolves(mutantResults);
+
+      cliOptions = {};
+      sut = new Stryker(cliOptions, () => injectorMock as typedInject.Injector<MutationTestContext>);
+    });
+
+    it('should not disable `removeDuringDisposal` on the temp dir when dry run rejects and cleanTempDir is set to `always`', async () => {
+      dryRunExecutorMock.execute.rejects(new Error('expected error for testing'));
+      await expect(sut.runMutationTest()).rejected;
+      expect(temporaryDirectoryMock.removeDuringDisposal).not.false;
+    });
+  });
```

Comment:
Why do you recreate an entire `describe` here? You should be able to just put this test in the previous `describe`:

```ts
  it('should not disable `removeDuringDisposal` on the temp dir when dry run rejects and cleanTempDir is set to `always`', async () => {
      testInjector.options.cleanTempDir= 'always';
      dryRunExecutorMock.execute.rejects(new Error('expected error for testing'));
      await expect(sut.runMutationTest()).rejected;
      expect(temporaryDirectoryMock.removeDuringDisposal).not.false;
    });
```

Reply:
Sorry for my blunder here, removed the describe section and I have a question why this line
`expect(temporaryDirectoryMock.removeDuringDisposal).not.false;`  when set to 
`expect(temporaryDirectoryMock.removeDuringDisposal).true;` throws `AssertionError: expected undefined to be true`

Reply:
Ah yes, this is because `temporaryDirectoryMock` is a mock, created on line 36. The property `removeDuringDisposal` is simply never set, so it doesn't get reflected here.

Reply:
I also added a unit test for the in `temporary-directory.spec.ts`: `"should remove the dir if cleanTempDir option is 'always'"`

Reply:
Thank for doing that.